### PR TITLE
feat(components/container): custom renderer for List column headers

### PR DESCRIPTION
**Pull request by romainseb**
**What is the problem this PR is trying to solve?**
We can't customize the List's column headers
**What is the chosen solution to this problem?**
Add a dictionary to customize it from outside
Add a first common header : the IconHeader
**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/VirtualizedList/HeaderIcon/HeaderIcon.component.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * This renderer render the header with an icon
+ * @param {object} props column header props
+ */
+function HeaderIcon({ label, sortBy, dataKey, sortDirection, columnData }) {
+	return (
+		<div className={classnames(theme['tc-header-icon'], 'tc-header-icon')}>
+			{columnData.iconName && <Icon name={columnData.iconName} title={label} />}
+			{sortBy === dataKey && <SortIndicator key="SortIndicator" sortDirection={sortDirection} />}
```
Comment (jmfrancois) [inline packages/components/src/VirtualizedList/HeaderIcon/HeaderIcon.component.js]: why do you need a key here ?

File: packages/components/src/VirtualizedList/utils/dictionary.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ import CellTitleRenderer, { cellType as cellTitleType } from '../CellTitle';
 import CellBadgeRenderer, { cellType as cellBadgeType } from '../CellBadge';
 import CellDatetimeRenderer, { cellType as cellDatetimeType } from '../CellDatetime';
 import CellTextIconRenderer, { cellType as cellTextType } from '../CellTextIcon';
+
```
Comment (jmfrancois) [inline packages/components/src/VirtualizedList/utils/dictionary.js]: please remove this empty line

File: packages/containers/src/List/List.container.js [hunk 3]
```diff
@@ -216,6 +216,19 @@ class List extends React.Component {
 			}, {});
 		}
 
+		console.log(props.headerDictionary);
```
Comment (jmfrancois) [inline packages/containers/src/List/List.container.js]: please remove this console.log

Comment (romainseb) [reply packages/containers/src/List/List.container.js]: Ohhhhh i haven't seen it, this need to disappear 🔥

File: packages/components/src/VirtualizedList/HeaderIcon/HeaderIcon.component.js (hunk 1 above)
Comment (romainseb) [reply packages/components/src/VirtualizedList/HeaderIcon/HeaderIcon.component.js]: I mirrored https://github.com/bvaughn/react-virtualized/blob/master/source/Table/defaultHeaderRenderer.js
As it return an array in the default headerRenderer, the key is required. As i don't return an array, you're right, i will remove it :)

File: packages/components/src/List/ListToVirtualizedList/ListToVirtualizedList.component.js [hunk 4]
```diff
@@ -94,6 +95,10 @@ export function ListToVirtualizedList(props) {
 					if (column.hideHeader) {
 						cProps.disableSort = true;
 						cProps.headerRenderer = HiddenHeader;
+					} else if (column.header && listHeaderDictionary[column.header]) {
+						Object.assign(cProps, listHeaderDictionary[column.header], {
```
Comment (jare-talend) [inline packages/components/src/List/ListToVirtualizedList/ListToVirtualizedList.component.js]: use spread istead of?

**Pull request merged**